File: code/cpp/src/topic_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

#include <iostream>


void callback(const std_msgs::Int32::ConstPtr &msg) {  // <1>
  std::cout << msg->data << std::endl;
}


int main(int argc, char **argv) {
  ros::init(argc, argv, "count_subscriber");
  ros::NodeHandle node;

  ros::Subscriber sub = node.subscribe("counter", 10, callback);  // <2>

  ros::spin();
}

File: code/basics/src/topic_subscriber.py
#!/usr/bin/env python
# BEGIN ALL
#!/usr/bin/env python


import rospy
from std_msgs.msg import Int32


# BEGIN CALLBACK
def callback(msg):
    print msg.data
# END CALLBACK


rospy.init_node('topic_subscriber')

# BEGIN SUBSCRIBER
sub = rospy.Subscriber('counter', Int32, callback)
# END SUBSCRIBER

rospy.spin()
# END ALL
